graph/src/blockchain/mod.rs

```diff
@@ -84,7 +84,7 @@ pub trait Blockchain: Debug + Sized + Send + Sync + Unpin + 'static {
     type TriggersAdapter: TriggersAdapter<Self>;
 
     /// Trigger data as parsed from the triggers adapter.
-    type TriggerData: TriggerData + Ord + Send + Sync;
+    type TriggerData: TriggerData + Ord + Send + Sync + Clone;
```

Comment:
Needing a `Clone` bound is always unfourtunate. Since this is only for tests, I'd prefer to remove it here and add `where C::TriggerData: Clone` bounds where necessary, to flag the places where cloning happens.